azure/services/aso/aso.go

```diff
@@ -291,3 +298,46 @@ func IsManaged(ctx context.Context, ctrlClient client.Client, spec azure.ASOReso
 func ownedByCluster(labels map[string]string, clusterName string) bool {
 	return labels[infrav1.OwnedByClusterLabelKey] == clusterName
 }
+
+// PauseResource pauses an ASO resource by updating its `reconcile-policy` to `skip`.
+func PauseResource(ctx context.Context, ctrlClient client.Client, spec azure.ASOResourceSpecGetter, clusterName string, serviceName string) error {
+	ctx, log, done := tele.StartSpanWithLogger(ctx, "aso.Service.Pause")
+	defer done()
+
+	resource := spec.ResourceRef()
+	resourceName := resource.GetName()
+	resourceNamespace := resource.GetNamespace()
+
+	log = log.WithValues("service", serviceName, "resource", resourceName, "namespace", resourceNamespace)
```

Comment:
Should this also be a `log.V(4)`?

Reply:
This doesn't log anything by itself, it only says that future calls to `log.Info/Error` will include those as fields. Since it looks like all the actual log calls below already use `.V(4)`, I suppose we could include that here. Then we'd need to refactor as soon as we change this to log something at a different verbosity though, so I think it might be a bit better to specify the verbosity on each log line.

Reply:
Ah nevermind, I read too fast and thought this was an actual logging call, not a setup func.